File: asn2/badtime.c
#include <stdlib.h>
#include <stdio.h>
#include <math.h>
#include <time.h>

void badsort(int *mylist,int n){
    if(n==2 && mylist[0]>mylist[1]){
        int temp = mylist[0];
        mylist[0] = mylist[1];
        mylist[1] = temp;
    }
    else if(n>2){
        float a = 0.666;
        int m = (int)(a*n);
        int *mysub = &mylist[0];
        badsort(mysub,m-1);
        mysub = &mylist[n-m];
        badsort(mysub,m);
        mysub = &mylist[0];
        badsort(mysub,m-1);
    }
}

int main(int argc,char *argv[]){
    srand(time(NULL));
    time_t time0, time1;
    int i;
    int size = atoi(argv[1]);
    int *randint = malloc(size+1*sizeof(int));
    for(i=0;i<size;i++){
        randint[i] = rand() % 10000;
    }
    clock(&time0);
    badsort(randint,size);
    clock(&time1);
    double mytime = (double)(time1-time0);
    printf("\nTime: %1.6f\n",time1);
    printf("\nsize: %d\n",size);
    for(i=0;i<size;i++){
        printf(" %d ",randint[i]);
    }
    printf("\n");
}

File: asn2/badtime.py
import time
import sys
import random

badsortnum = 0

def badsort(myline,n):
    global badsortnum
    badsortnum += 1
    if len(myline) == 2 and myline[0] > myline[1]:
        temp = myline[0]
        myline[0] = myline[1]
        myline[1] = temp
    elif len(myline) > 2:
        m = n*len(myline)
        #print(str(m))
        #bad1 = badsort(myline[0:int(m-1)],n)
        #bad1.extend(myline[int(m):len(myline)])
        #myline = bad1
        #print(myline)
        #bad2 = myline[0:int(len(myline)-m-1)]
        #bad2.extend(badsort(myline[int(len(myline)-m):len(myline)-1],n))
        #myline = bad2
        #print(myline)
        #bad3 = badsort(myline[0:int(m-1)],n)
        #bad3.extend(myline[int(m):len(myline)])
        #myline = bad3
        badsort(myline[0:int(m-1)],n)
        badsort(myline[int(len(myline)-m):len(myline)-1],n)
        badsort(myline[0:int(m-1)],n)
    return myline

def brutesort(myline):
    maxsum = 0
    for i in range(0,len(myline)):
        mysum = 0
        for j in range(i,len(myline)):
            mysum += myline[j]
            if mysum > maxsum:
                maxsum = mysum
                low = i
                high = j
    high += 1
    return myline[low:high]
    

def main():
    #init files
    global badsortnum
    outputfile = open("badsort.out","w")
    times = open("badtime.out","w")
    #generate random list of numbers
    myline = list()
    size = int(sys.argv[1])
    for i in range(0,size):
        myline.append(random.randint(0,10000))
    #run test
    #print(myline)
    time0 = time.time()
    #mylist = brutesort(myline)
    mylist = badsort(myline,0.66)
    time1 = time.time() - time0
    print(str(badsortnum))
    #print the results to file
    for i in mylist:
        outputfile.write(str(i))
        outputfile.write(" ")
    outputfile.write("\n")
    times.write("Sample size: ")
    times.write(str(size))
    times.write("\nTime: ")
    times.write(str(time1))
    times.write("\n")
    times.close()
    outputfile.close()

if __name__ == '__main__':
    main()